### ISSUE TITLE
✨ Add Conformer CTC Pretraining Support for New Languages

### ISSUE THREAD

**Issue by mojtaba-nafez** [ibm-granite/granite-speech-models#3]
Current Limitation
The existing notebook fine-tunes only the Q-Former and LoRA adapters, omitting Conformer encoder pretraining with CTC loss (per the [Granite-Speech paper](https://arxiv.org/abs/2405.07593)).

Request
Please add:

- 📜 Script/notebook for Conformer CTC pretraining on ASR datasets.
- 🔤 Tokenizer guidance (char vs. subword setup).
- ⚙️ Example hyperparameters (e.g., 10 layers, LR, batch size).
- 🔗 Instructions to integrate the pretrained encoder into the Q-Former + LLM pipeline.

Why It Matters
Pretraining the Conformer in-language is essential to match Whisper-like performance and enable full pipeline replication. My initial fine-tuning showed promise, but encoder pretraining is needed for competitive accuracy.

Thank you for your impactful work—this open-source release is greatly appreciated! 🙌

**Comment by msamwelmollel** [comment]
Add something like this to your code 

for n, p in model.named_parameters():
    total_params += p.numel()
    
    # Conservative selection based on actual layer names
    if any(keyword in n.lower() for keyword in [
        # Core components (always include)
        "lora_a", "lora_b",           # LoRA adapters: lora_A.default.weight, lora_B.default.weight
        
        # Speech encoder (critical for audio processing)
        "encoder.",                   # All encoder layers: encoder.layers.0-15
        
        # Projector components (speech to language bridge)
        "projector.query",            # Query embeddings
        "projector.qformer",          # Query transformer (CRITICAL!)
        "projector.linear",           # Final projection
        
        # Normalization layers (stability)
        "layernorm", "layer_norm",    # All normalization
        "post_attention_layernorm",   # Post-attention norms
        "input_layernorm",            # Input norms
        "bias",                       # All bias terms
        
        # Language model - last 3 layers only
        "language_model.model.layers.39",  # Very last layer
        "language_model.model.layers.38",  # Second to last
        "language_model.model.layers.37",  # Third to last
        "language_model.model.layers.36",  # option etc 
        "language_model.model.layers.35",  # option etc 
        "language_model.model.layers.34",  # option etc 
        "language_model.model.layers.33",  # option etc 
        "language_model.model.layers.32",  # option etc 
        "language_model.model.layers.31",  # option etc 
        "language_model.model.layers.30",  # option etc 
        
        # Add attention mechanisms in trained layers
        "self_attn.q_proj", "self_attn.v_proj",  # Only q and v (have LoRA)
        "self_attn.o_proj",                      # Output projection
    ]):
        p.requires_grad = True
        trainable_params += p.numel()
        print(f"✓ Training: {n}")
    else:
        p.requires_grad = False


You can add as many layers as possible
